### PR TITLE
MM-1617 Now shows a loading gif while a profile pic is being uploaded

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Uses the loading gif from file preview/uploading to replace the save button while a profile picture is being uploaded.  Also included minor cosmetic refactoring to better fit style guide for the "general tab" of user settings.

### CODE REVIEW

**Comment by crspeller** [comment]
Looks good to me. But needs to be rebased. @rgarmsen2295

File: web/react/components/setting_picture.jsx [hunk 1]
```diff
@@ -41,10 +41,14 @@ module.exports = React.createClass({
                             {img}
                         </li>
                         <li className='setting-list-item'>
-                            {server_error}
-                            {client_error}
+                            {serverError}
+                            {clientError}
                             <span className='btn btn-sm btn-primary btn-file sel-btn'>Select<input ref='input' accept='.jpg,.png,.bmp' type='file' onChange={this.props.pictureChange}/></span>
+                            {!this.props.loadingPicture ?
```
Comment (nickago) [inline web/react/components/setting_picture.jsx]: We probably shouldn't be adding any ternary operators to the code if we're just going to need to take them out

**Comment by hmhealey** [comment]
Looks good other than the ternary statements as Nick pointed out.

File: web/react/components/setting_picture.jsx (hunk 1 above)
Comment (rgarmsen2295) [inline web/react/components/setting_picture.jsx]: Agreed, didn't even think about it (or see the red apparently) thanks taking it out now.

**Comment by rgarmsen2295** [comment]
Ok completed the additional refactoring, please take another look when possible @nickago @hmhealey thanks

**Comment by nickago** [comment]
+1

**Pull request merged**